versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java

```diff
@@ -32,8 +34,9 @@ public interface ChangeRecorder {
      *
      * @param changeRecord a record described change
      * @since 2.14.0
+     * @param <T> concrete {@link VersionChange} sub-interface
      */
-    void recordChange(ChangeRecord changeRecord);
+    <T extends VersionChange> void recordChange(ChangeRecord<T> changeRecord);
```

Comment:
What do you think add two methods

```
void recordChange(DependencyChangeRecord changeRecord);

void recordChange(PropertyChangeRecord changeRecord);
```

Implementation will be easier - we will not need to use `instance of`

Reply:
Well, this has its downsizes too -- this means a change in the interface.

Reply:
I'll remove the generic void recordChange(ChangeRecord changeRecord);

Reply:
Applied; not squashing yet, waiting for final review.